core/ui/src/main/kotlin/org/meshtastic/core/ui/timezone/ZoneIdExtensions.kt

```diff
@@ -109,15 +110,15 @@ private fun ZonedDateTime.formattedOffsetString(): String {
 }
 
 /**
- * Returns a date string with the format ",M<MONTH>.<WEEK_OF_MONTH>.<WEEKDAY>/<HOUR>:<MINUTE>:<SECOND>". Time is omitted
- * if it is 2:00:00, since that is the default. Otherwise, append time with non-zero values.
+ * Returns a date string with the format ",M<MONTH>.<WEEK_OF_MONTH>.<DAY_OF_WEEK>/<HOUR>:<MINUTE>:<SECOND>". Time is
+ * omitted if it is 2:00:00, since that is the default. Otherwise, append time with non-zero values.
  */
 private fun LocalDateTime.formattedDateString(): String {
     @Suppress("MagicNumber")
-    val weekFields = WeekFields.of(DayOfWeek.SUNDAY, 7)
-
     return buildString {
-        append(",M$monthValue.${get(weekFields.weekOfMonth())}.${get(weekFields.dayOfWeek()) - 1}")
+        val weekOfMonth = get(ChronoField.ALIGNED_WEEK_OF_MONTH)
```

Comment:
I wasn't able to find a "weekday ordinal" (e.g. 4th Friday) equivalent in `java.time` (though it may exist somewhere), but this should be functionally the same. `ALIGNED_WEEK_OF_MONTH` ensures that months are broken into 7 day chunks starting on the first day of the month (and not the partial week of the previous month), meaning that "week number" should be functionally the same as "weekday ordinal".

This fixes parsing for more edge rules like `Africa/Cairo` that occur both late in the week _and_ late in the month.